pkg/controllers/dynakube/logmodule/daemonset/reconciler_test.go

```diff
@@ -0,0 +1,276 @@
+
+		reconciler := NewReconciler(mockK8sClient, mockK8sClient, dk)
+		err := reconciler.Reconcile(ctx)
+
+		require.NoError(t, err)
+		assert.Empty(t, *dk.Conditions())
+
+		var daemonset appsv1.DaemonSet
+		err = mockK8sClient.Get(ctx, types.NamespacedName{
+			Name:      GetName(dk.Name),
+			Namespace: dk.Namespace,
+		}, &daemonset)
+		require.True(t, k8serrors.IsNotFound(err))
+	})
+
+	t.Run("problem with k8s request => visible in conditions", func(t *testing.T) {
+		dk := createDynakube(true)
+
+		boomClient := createBOOMK8sClient()
+
+		reconciler := NewReconciler(boomClient,
+			boomClient, dk)
+
+		err := reconciler.Reconcile(context.Background())
+
+		require.Error(t, err)
+		require.Len(t, *dk.Conditions(), 1)
+		condition := meta.FindStatusCondition(*dk.Conditions(), conditionType)
+		assert.Equal(t, conditions.KubeApiErrorReason, condition.Reason)
+		assert.Equal(t, metav1.ConditionFalse, condition.Status)
+	})
+}
+
+func TestGenerateDaemonSet(t *testing.T) {
+	t.Run("generate daemonset", func(t *testing.T) {
+		dk := createDynakube(true)
+
+		reconciler := NewReconciler(nil,
+			nil, dk)
+		daemonset, err := reconciler.generateDaemonSet()
+		require.NoError(t, err)
+		require.NotNil(t, daemonset)
+
+		assert.Len(t, daemonset.Spec.Template.Spec.InitContainers, 1)
+		assert.Len(t, daemonset.Spec.Template.Spec.Containers, 1)
+		assert.NotEmpty(t, daemonset.Spec.Template.Spec.Volumes)
+		assert.Equal(t, GetName(dk.Name), daemonset.Name)
+		assert.Equal(t, dk.Namespace, daemonset.Namespace)
+		assert.NotEmpty(t, daemonset.Labels)
+		assert.NotEmpty(t, daemonset.Spec.Template.Labels)
+		assert.NotEmpty(t, daemonset.Spec.Template.Spec.Affinity)
+		assert.Subset(t, daemonset.Spec.Template.Labels, daemonset.Spec.Selector.MatchLabels)
+		require.Len(t, daemonset.Annotations, 1)
+		assert.Contains(t, daemonset.Annotations, hasher.AnnotationHash)
+		assert.Equal(t, serviceAccountName, daemonset.Spec.Template.Spec.ServiceAccountName)
+		assert.Empty(t, daemonset.Spec.Template.Spec.DNSPolicy)
+		assert.Empty(t, daemonset.Spec.Template.Spec.PriorityClassName)
+		assert.Empty(t, daemonset.Spec.Template.Spec.Tolerations)
+		assert.Len(t, daemonset.Spec.Template.Spec.ImagePullSecrets, 1)
+		require.NotNil(t, daemonset.Spec.UpdateStrategy.RollingUpdate)
+		assert.Equal(t, intstr.FromInt(dk.FeatureOneAgentMaxUnavailable()), *daemonset.Spec.UpdateStrategy.RollingUpdate.MaxUnavailable)
+	})
+
+	t.Run("respect custom labels", func(t *testing.T) {
+		customLabels := map[string]string{
+			"custom": "label",
+		}
+
+		dk := createDynakube(true)
+		dk.Spec.Templates.LogModule.Labels = customLabels
+
+		reconciler := NewReconciler(nil,
+			nil, dk)
+		daemonset, err := reconciler.generateDaemonSet()
+		require.NoError(t, err)
+		require.NotNil(t, daemonset)
+
+		assert.Subset(t, daemonset.Spec.Template.Labels, customLabels)
+		// assert.Subset(t, daemonset.Labels, customLabels) TODO: Should this be the case?
```

Comment:
should it? Checking the activegate, custom labels are only put on the pods

Reply:
that is why I am asking :D

Reply:
let's keep consistency with other components